posthog/api/test/test_insight.py

```diff
@@ -1366,6 +1366,28 @@ def test_another_user_viewing_an_insight_does_not_impact_the_list(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response_data["results"]), 0)
 
+    def test_get_recent_insights_with_feature_flag(self):
+        filter_dict = {"events": [{"id": "$pageview"}], "breakdown": "$feature/insight-with-flag-used"}
+        filter_dict2 = {
+            "events": [{"id": "$pageview"}],
+            "properties": [{"key": "$active_feature_flag", "value": "insight-with-flag-used"}],
+        }
+
+        insight = Insight.objects.create(
```

Comment:
with tests, it's nice to have both a positive & negative situation, so you know that the filter is doing exactly what you expect.

For example, if the `filter` here were doing nothing at all, this test would still pass, which isn't great.

```suggestion
        insight = Insight.objects.create(
            filters=filter_dict.with_data({"breakdown": "email"}.to_dict(), team=self.team, short_id="123"
        )
        insight = Insight.objects.create(
```

Reply:
i'm confused, if you're using an incorrect filter this test is passing?

Reply:
Exactly my point.

Imagine if the code said:

```
elif key == "feature_flag":
                feature_flag = request.GET["feature_flag"]
                queryset = queryset
```

Then, this test would pass right, since there's 2 total insights, and you're testing that the API returns 2.

But after my change, this test won't pass, since it will return 3, when it should return 2.